libraries/AP_HAL_ChibiOS/Scheduler.cpp

```diff
@@ -666,6 +666,26 @@ void Scheduler::restore_interrupts(void *state)
     chSysRestoreStatusX((syssts_t)(uintptr_t)state);
 }
 
+struct task_ctx {
```

Comment:
You could add a destructor to this to clean up the two members and then use new and delete in the usage - might make it slightly cleaner

Reply:
> You could add a destructor to this to clean up the two members and then use new and delete in the usage - might make it slightly cleaner

If I'm understanding your suggestion correctly that would involve moving doing this allocation into the constructor for `struct task_ctx` here?

Couldn't *quite* do that as you can't fail in a constructor.  But I've pushed some patches on top which add an initialise method, and yes, I think it is an improvement, thanks!

Reply:
.... I've now removed the calls to malloc in here.  I've no idea what I was thinking when I added those, but it seems to get by with just having the init/body procs as members of the context.

I've done a smoke-test on a CubeOrange here and it seems to do logging just fine-and-dandy.